### PR TITLE
Merge GemVersionPromoter into outdated cmd

### PR DESCRIPTION
**Pull request by ryanfox1985**
Hi I created this PR according with: 
https://github.com/bundler/bundler/pull/4841

Regards.

### CODE REVIEW

**Comment by chrismo** [comment]
thx - i actually had started working on this locally, i'm not sure some of these changes are going to be necessary, but we'll see.

**Comment by ryanfox1985** [comment]
@chrismo, If you want I can apply your renames.

**Comment by ryanfox1985** [comment]
@chrismo, can you do the review here? 
I can explain the code.

Regards.

**Comment by ryanfox1985** [comment]
I think that I need to fix one test and rollback the changes on `definition.rb` (because I don't need).

**Comment by chrismo** [comment]
@ryanfox1985 I ended up taking care of this in #5061, unless you see something there I missed, we can close yours.

File: lib/bundler/cli/outdated.rb [hunk 1]
```diff
@@ -47,31 +50,41 @@ def run
...
+          if !current_spec.version.prerelease? && !options[:pre] && specs.size > 1
+            specs.delete_if {|b| b.respond_to?(:version) && b.version.prerelease? }
+          else
+            specs
           end
-          active_spec = active_specs.last
         end
 
-        if options["filter-major"] || options["filter-minor"] || options["filter-patch"]
+        active_spec = Bundler.definition.gem_version_promoter.sort_versions(dependency, spec_groups).last
```
Comment (ryanfox1985) [inline lib/bundler/cli/outdated.rb]: @chrismo In master code is not calling to `gem_version_promoter.sort_versions`. I think it's a good idea keep calling to `gem_version_promoter`.

I created another PR https://github.com/bundler/bundler/pull/5114 for clean code stuff (I will remove the refactors from this PR).

**Comment by chrismo** [comment]
closing since this was taken care of in https://github.com/bundler/bundler/pull/5061

**Pull request closed without merging**